src/components/Tables/Table.tsx

```diff
@@ -0,0 +1,149 @@
+"use client";
+
+import React, { useState, useEffect } from "react";
+import DataTable, { TableColumn } from "react-data-table-component";
+import TableFilterBar, { FilterText } from "@/components/Tables/TableFilterBar";
+
+interface Row {
+    checkBoxKey: number;
```

Comment:
potentially better as `rowId`, which describes what the index _is_ rather than what it _does_

Reply:
the idea of row id might be a bit conflicting with the parameter id
id describes after filtering, which specific row it is located in (which is called in the bottom as row.id for each row object)
for checkBoxKey, it is specifically just a unique identifier of each row object that is the same as the unfiltered data (props.data) index, it is irrelevant to which row a particular datum is in the table hence we avoid doing so

Reply:
nevermind, resolved as suggested